services/src/main/java/org/fao/geonet/guiapi/search/XsltResponseWriter.java

```diff
@@ -69,6 +69,7 @@ public XsltResponseWriter(String envTagName, String serviceName) {
         gui.addContent(new Element("nodeUrl").setText(settingManager.getNodeURL()));
         gui.addContent(new Element("baseUrl").setText(settingManager.getBaseURL()));
         gui.addContent(new Element("serverUrl").setText(settingManager.getServerURL()));
+        gui.addContent(new Element("nodeId").setText(settingManager.getNodeId()));
```

Comment:
Change looks fine bug if you look at #7015 you will see the following changes to xsltFormatter.java

https://github.com/geonetwork/core-geonetwork/pull/7015/files#diff-8cece58729f579a61fbbba7e3e0f30b216a26d0c960abc5fd3a74af210681556

https://github.com/ianwallen/core-geonetwork/blob/ef58ccfd7bd3d2b3947432847c7eadc29b9352f2/services/src/main/java/org/fao/geonet/api/records/formatters/XsltFormatter.java#L121-L145

Without these changes, it would not load the translation text and it would not display the page correctly.  Have you tested the translation text?

Reply:
@ianwallen I doubt this works as I see the code sending the English json files hardcoded:

https://github.com/geonetwork/core-geonetwork/blob/e8dda2e8de2a9af9f629a56f4ffc515520dfe8dc/services/src/main/java/org/fao/geonet/guiapi/search/SearchApi.java#L106-L107

I would create a ticket to improve this and also check how to implement the API service in GeoNetwork 4.x.

But for now I would like to merge the current changes as in certain setups in 3.12.x causes an error in the service.

Reply:
That is fine - I just wanted to make sure that you were aware of the translation issues